File: Cpp/1260_Shift_2D_Grid.cpp
class Solution {
public:
    vector<vector<int>> shiftGrid(vector<vector<int>>& grid, int k) {
        int m = grid.size();
        int n = grid[0].size();
        vector<vector<int>> result(m, vector<int>(n, 0));
        for(int i=0; i<m; ++i){
            for(int j=0; j<n; ++j){
                int row = (i+(j+k)/n)%m;
                int col = (j+k)%n;
                result[row][col] = grid[i][j];
            } 
        }
        
        return result;
    }
};

File: Kotlin/1260_Shift_2D_Grid.kt
class Solution {
    fun shiftGrid(grid: Array<IntArray>, k: Int): List<List<Int>> {
        val m = grid.size
        val n = grid[0].size
        val result = MutableList(m) {MutableList(n) {0}}

        for(i in 0 until m){
            for(j in 0 until n){
                val row = (i + (j + k) / n) % m
                val col = (j + k) % n
                result[row][col] = grid[i][j]
            } 
        }
        
        return result
    }
}